lib/galaxy/tools/parameters/basic.py

```diff
@@ -1517,7 +1517,7 @@ def get_options(self, trans, other_values):
                 except Exception:
                     column_list = self.get_column_list(trans, other_values)
             if self.numerical:  # If numerical was requested, filter columns based on metadata
-                if hasattr(dataset, "metadata") and hasattr(dataset.metadata, "column_types"):
+                if hasattr(dataset, "metadata") and hasattr(dataset.metadata, "column_types") and column_list is not None:
```

Comment:
In https://sentry.galaxyproject.org/organizations/galaxy/issues/144649/?project=3 it's not `column_list` that is None, and by looking at the code above I don't think it can actually be `None`. So the error was likely that `dataset.metadata.column_types` is None, as can happen for example for the CMAP datatype in `lib/galaxy/datatypes/tabular.py` .

```suggestion
                if hasattr(dataset, "metadata") and hasattr(dataset.metadata, "column_types") and dataset.metadata.column_types is not None:
```

It would probably be good to also fix the CMAP datatype with:

```diff
diff --git a/lib/galaxy/datatypes/tabular.py b/lib/galaxy/datatypes/tabular.py
index fab0b58933..68e64fcce9 100644
--- a/lib/galaxy/datatypes/tabular.py
+++ b/lib/galaxy/datatypes/tabular.py
@@ -1814,7 +1814,7 @@ class CMAP(TabularData):
             with open(dataset.get_file_name()) as dataset_fh:
                 comment_lines = 0
                 column_headers = None
-                cleaned_column_types = None
+                cleaned_column_types = []
                 number_of_columns = 0
                 for i, line in enumerate(dataset_fh):
                     line = line.strip("\n")
@@ -1822,7 +1822,6 @@ class CMAP(TabularData):
                         if line.startswith("#h"):
                             column_headers = line.split("\t")[1:]
                         elif line.startswith("#f"):
-                            cleaned_column_types = []
                             for column_type in line.split("\t")[1:]:
                                 if column_type == "Hex":
                                     cleaned_column_types.append("str")
```

Reply:
Might as well do `if hasattr(dataset, "metadata") and getattr(dataset.metadata, "column_types", None) is not None:`

Reply:
> Might as well do `if hasattr(dataset, "metadata") and getattr(dataset.metadata, "column_types", None) is not None:`

Yes, that's better and shorter, so should fix the Python linting failure as well.